### PR TITLE
br-stream: backup record key only

### PR DESCRIPTION
**Pull request by YuJuncen**
Signed-off-by: Yu Juncen <yu745514916@live.com>

### What problem does this PR solve?

Issue Number: ref #32167

Problem Summary: See the issue.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

```release-note
None
```

### CODE REVIEW

**Review by joccau — APPROVED** [review]
LGTM

File: br/pkg/task/stream.go [hunk 1]
```diff
@@ -676,9 +676,19 @@ func RunStreamRestore(
...
 	}
 
-	// TODO split put and delete files
 	// perform restore kv files
-	return client.RestoreKVFiles(ctx, rewriteRules, datas)
+	if err := client.RestoreKVFiles(ctx, rewriteRules, datas); err != nil {
+		return errors.Trace(err)
+	}
+
+	for _, table := range fullBackupTables {
```
Comment (3pointer) [inline br/pkg/task/stream.go:684]: maybe fullBackupTables's table info is outdate?

File: br/pkg/restore/client.go [hunk 2]
```diff
@@ -1251,6 +1251,43 @@ func (rc *Client) ReadStreamDataFiles(ctx context.Context, metas []*backuppb.Met
 	return streamBackupDataFiles, nil
 }
 
+// FixIndex tries to fix a single index.
+func (rc *Client) FixIndex(ctx context.Context, schema, table, index string) error {
+	if span := opentracing.SpanFromContext(ctx); span != nil && span.Tracer() != nil {
+		span1 := span.Tracer().StartSpan(fmt.Sprintf("Client.LoadRestoreStores index: %s",
+			strings.Join([]string{schema, table, index}, ".")), opentracing.ChildOf(span.Context()))
```
Comment (kennytm) [inline br/pkg/restore/client.go:1258]: ok why not just make it `fmt.Sprintf("Client.LoadRestoreStores index: %s.%s.%s", schema, table, index)`

File: br/pkg/task/stream.go (hunk 1 above)
Comment (YuJuncen) [reply br/pkg/task/stream.go:684]: I guess that was out of our scope 🤔. If the table list is stale, `RestoreKVFiles` would skip those tables too. For handling incremental tables, we should apply the meta key change at the `fullBackupTables`(Which hasn't been implemented for now).

**Review by kennytm — APPROVED** [review]

**Comment by YuJuncen** [comment]
/run-mysql-test 
Cancelling nested steps due to timeout 🤔

**Comment by YuJuncen** [comment]
The branch is too old to pass the `mysql_test`... Merge it manually...

**Pull request merged**